src/java/test/org/apache/zookeeper/server/quorum/QuorumPeerMainTest.java

```diff
@@ -470,16 +481,20 @@ public void testBadPeerAddressInQuorum() throws Exception {
     @Test
     public void testInconsistentPeerType() throws Exception {
         ClientBase.setupTestEnv();
-
-        // setup the logger to capture all logs
+        
+        LoggerContext loggerContext =  (LoggerContext) LogManager.getContext(false);
+        Configuration logConfig = loggerContext.getConfiguration();
+        
         Layout layout =
-                Logger.getRootLogger().getAppender("CONSOLE").getLayout();
+        		logConfig.getRootLogger().getAppenders().get("STDOUT").getLayout();
+        String appenderName = "quorumAppender";
         ByteArrayOutputStream os = new ByteArrayOutputStream();
-        WriterAppender appender = new WriterAppender(layout, os);
-        appender.setThreshold(Level.INFO);
-        Logger qlogger = Logger.getLogger("org.apache.zookeeper.server.quorum");
-        qlogger.addAppender(appender);
-
+        WriterAppender appender = WriterAppender.createAppender((StringLayout)layout, null, new OutputStreamWriter(os), appenderName, false, true);
```

Comment:
It's probably a good idea to use the builder to create the appender (see above).